### PR TITLE
refactor: Add caching to KnoraGroupRepo (DEV-3311)

### PR DESCRIPTION
**Pull request by seakayone**
- **refactor: Add caching to KnoraGroupRepo**
- **Removes last remaining code in GroupsResponder updating a group, all group updates now go through the repo save mehtod**

### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)
- [ ] deprecated: Deprecation warning (ideally referencing a migration guide)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala [hunk 1]
```diff
@@ -75,19 +78,23 @@ final case class GroupRestService(
...
-      groupToUpdate <- groupService
-                         .findById(iri)
-                         .someOrFail(NotFoundException(s"Group <${iri.value}> not found."))
-      internal <- groupService.updateGroupStatus(groupToUpdate, request.status).map(GroupGetResponseADM.apply)
+      groupAndProject <- auth.ensureSystemAdminOrProjectAdminOfGroup(user, iri)
+      (group, _)       = groupAndProject
+      internal <- knoraGroupService
+                    .updateGroupStatus(group, request.status)
+                    .flatMap(groupService.toGroup)
+                    .map(GroupGetResponseADM.apply)
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala:86]: nitpick: not sure if it makes sense to have for-comprehensions if we use map and flatmap in them; or vice versa, if it makes sense to use map and flatmap, if we're in a for-comprehension

**Review by BalduinLandolt — APPROVED** [review]
Looks good, thanks!

File: webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala [hunk 2]
```diff
@@ -395,14 +395,12 @@ final case class AuthenticatorLive(
                   case Some(KnoraJWTTokenCredentialsV2(jwtToken)) =>
                     ZIO
                       .fail(BadCredentialsException(BAD_CRED_NOT_VALID))
-                      .whenZIO(jwtService.validateToken(jwtToken).map(!_))
+                      .when(!jwtService.isTokenValid(jwtToken))
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala:398]:
```suggestion
                      .unless(jwtService.isTokenValid(jwtToken))
```

File: webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala [hunk 3]
```diff
@@ -395,14 +395,12 @@ final case class AuthenticatorLive(
...
                       .as(true)
                   case Some(KnoraSessionCredentialsV2(sessionToken)) =>
                     ZIO
                       .fail(BadCredentialsException(BAD_CRED_NOT_VALID))
-                      .whenZIO(
-                        jwtService.validateToken(sessionToken).map(!_),
-                      )
+                      .when(!jwtService.isTokenValid(sessionToken))
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/routing/Authenticator.scala:403]:
```suggestion
                      .unless(jwtService.isTokenValid(sessionToken))
```

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala (hunk 1 above)
Comment (mpro7) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala:86]: IMO this still reads better than chaining it all in one go. These lines could be also extracted to separate private methods, especially they are reused in `deleteGroup` methods below.

**Review by mpro7 — APPROVED** [review]
Thanks. Please find my minor suggestions below:

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala]: 1. I am totally ok to mix for comprehension with a flatmap here and there,
2. I am totally ok with a chain of flatmaps here and there
4. I had my reason of why the internal has become its own name and the intermediate transformative results where in a chain of map/flatMap for my personal preference of making it clear that toGroup and Foo.apply are simply transformations of intermediate unimportant results.
5. I have extracted the code duplication in a separate method and introduced intermediate result name due to popular demand. [b247906](https://github.com/dasch-swiss/dsp-api/pull/3204/commits/b247906d4777dc9fe9da175e9d0ac1c5aaefd5f7)

TBH, I find this kind of discussion exhausting and have a different preference for readability.

Comment (siers) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala]: It does look better with the flatMap refactored.

**Pull request merged**